ouroboros-consensus-test/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -77,6 +92,217 @@ tests = testGroup "Mempool"
   ]
 
+{-------------------------------------------------------------------------------
+  Mempool Pure Properties
+-------------------------------------------------------------------------------}
+
+-- | Test that removed transactions are not in the
+--   result transactionsIds of the returned internal state.
+prop_pure_removeTxs
+  :: TestSetupSub
+  -> Property
+prop_pure_removeTxs TestSetupSub {..} =
+    withInternalState tssSetup $
+    \ mpArgs internalSt ledgerState ->
+      let (internalStRes, _)    = pureRemoveTxs (map txId tssSubTxs) mpArgs internalSt ledgerState
+          txIdsRemaining        = Set.toList (isTxIds internalStRes)
+      in property $ null $ intersect txIdsRemaining (map txId tssSubTxs)
+
+-- | Test that after removing transactions all the other transactions are still available.
+--
+prop_pure_removeTxs_contra
+    :: TestSetupWithTxs
+    -> Property
+prop_pure_removeTxs_contra TestSetupWithTxs {..} =
+    withInternalState testSetup $
+    \ mpArgs internalSt ledgerState ->
+      let (_res,internalSt')     = runTryAddTxs mpArgs internalSt (map fst txs)
+          (internalStRes, _tra)  = pureRemoveTxs (map (txId . fst) txs)
+                                      mpArgs internalSt' ledgerState
+          txIdsRemaining         = Set.toList (isTxIds internalStRes)
+      in sort txIdsRemaining === sort (Set.toList (isTxIds internalSt))
+
+-- | Test that @snapshotTxs == snapshotTxsAfter zeroIdx@.
+prop_pure_snapshotTxs_snapshotTxsAfter :: TestSetup -> Property
+prop_pure_snapshotTxs_snapshotTxsAfter setup =
+    withInternalState setup $
+    \ _mpArgs internalSt _ledgerState ->
+      let MempoolSnapshot { snapshotTxs, snapshotTxsAfter } = implSnapshotFromIS internalSt
+      in  snapshotTxs === snapshotTxsAfter zeroTicketNo
+
+
+-- | Test that all valid transactions added to a 'Mempool' can be retrieved
+-- afterward.
+prop_pure_addTxs_getTxs
+  :: TestSetupWithTxs
+  -> Property
+prop_pure_addTxs_getTxs setup@TestSetupWithTxs {..} =
+  withInternalState testSetup $
+  \ mpArgs internalSt _ledgerState ->
+    let (_res,internalSt') = runTryAddTxs mpArgs internalSt (map fst txs)
+        MempoolSnapshot { snapshotTxs } = implSnapshotFromIS internalSt'
+    in counterexample (ppTxs txs) $
+          validTxs setup `isSuffixOf` map fst snapshotTxs
+
+-- | Same as 'prop_Mempool_addTxs_getTxs', but add the transactions one-by-one
+-- instead of all at once.
+prop_pure_addTxs_one_vs_multiple :: TestSetupWithTxs -> Property
+prop_pure_addTxs_one_vs_multiple setup@TestSetupWithTxs {..} =
+  withInternalState testSetup $
+  \ mpArgs internalSt _ledgerState ->
+      let internalSt' = foldl'  (\ is tx -> snd $ runTryAddTxs mpArgs is  [tx])
+                                internalSt
+                                (map fst txs)
+          MempoolSnapshot { snapshotTxs } = implSnapshotFromIS internalSt'
+      in counterexample (ppTxs txs) $
+          validTxs setup `isSuffixOf` map fst snapshotTxs
```

Comment:
I was expecting to see `runTryAddTxs mpArgs is txs` here so that we can clearly see what we're testing against.

Reply:
I don't understand your point. runTryAddTxs is called, and the test is that all valid transactions are added correctly.